examples/02_decoding/plot_haxby_understand_decoder.py

```diff
@@ -0,0 +1,326 @@
+"""Understanding :class:`nilearn.decoding.Decoder`
+==================================================
+
+Nilearn's :class:`nilearn.decoding.Decoder` object is a composite estimator
+that does several things under the hood and can hence be a bit difficult to
+understand at first.
+
+This example aims to provide a clear understanding of the
+:class:`nilearn.decoding.Decoder` object by demonstrating these steps via a
+Scikit-Learn pipeline.
+
+We will use the :footcite:t:`Haxby2001` dataset where the participants were
+shown images of 8 different types as described in the
+:ref:`sphx_glr_auto_examples_02_decoding_plot_haxby_anova_svm.py` example.
+We will train a classifier to predict the label of the object in the stimulus
+image based on the subject's fMRI data from the Ventral Temporal cortex.
+"""
+
+# %%
+# Load the Haxby dataset
+# ----------------------
+from nilearn import datasets
+
+# By default 2nd subject data will be fetched on which we run our analysis
+haxby_dataset = datasets.fetch_haxby()
+fmri_img = haxby_dataset.func[0]
+# Pick the mask that we will use to extract the data from Ventral Temporal
+# cortex
+mask_vt = haxby_dataset.mask_vt[0]
+
+# Load the behavioral data
+import pandas as pd
+
+from nilearn.image import index_img
+
+behavioral_data = pd.read_csv(haxby_dataset.session_target[0], sep=" ")
+labels = behavioral_data["labels"]
+# Keep the trials corresponding to all the labels except the ``rest`` ones
+labels_mask = labels != "rest"
+y = labels[labels_mask]
+y = y.values
+
+# Load run information
+run = behavioral_data["chunks"][labels_mask]
+run = run.values
+
+# Also keep the fmri data corresponding to these labels
+fmri_img = index_img(fmri_img, labels_mask)
+
+# Overview of the input data
+import numpy as np
+
+n_labels = len(np.unique(y))
+
+print(f"{n_labels} labels to predict (y): {np.unique(y)}")
+print(f"fMRI data shape (X): {fmri_img.shape}")
+print(f"Runs (groups): {np.unique(run)}")
+# %%
+# Preprocessing
+# -------------
+#
+# As we can see, the fMRI data is a 4D image with shape (40, 64, 64, 864).
+# Here 40x64x64 are the dimensions of the 3D brain image and 864 is the number
+# of selected trials, each corresponding to one of the 8 labels we selected
```

Comment:
More exactly,  it corresponds to the number of brain volumes acquired while visual stimuli were presented.